modules/ai/assets/js/editor/components/prompt-history/history-modal.js

```diff
@@ -0,0 +1,161 @@
+
+		lastRun.current();
+	}, [] );
+
+	const loadNext = () => {
+		if ( isLoading || isLastPage ) {
+			return;
+		}
+
+		lastRun.current = async () => fetchData( { page: meta.currentPage + 1, limit: ITEMS_LIMIT } );
+
+		lastRun.current();
+	};
+
+	const onHistoryItemDelete = async ( id ) => {
+		lastRun.current = async () => await deleteItem( id );
+
+		await lastRun.current();
+
+		if ( ! historyDeletingError ) {
+			deleteItemById( id );
+		}
+	};
+
+	const renderPeriods = () => {
+		const groupData = groupPromptHistoryData( items );
+		const periods = [];
+
+		if ( groupData[ LAST_7_DAYS_KEY ]?.items?.length ) {
+			periods.push( <PromptHistoryPeriod periodTitle={ groupData[ LAST_7_DAYS_KEY ].label }
+				onHistoryItemDelete={ onHistoryItemDelete }
+				historyItems={ groupData[ LAST_7_DAYS_KEY ].items } /> );
+		}
+
+		if ( groupData[ LAST_30_DAYS_KEY ]?.items?.length ) {
+			periods.push( <PromptHistoryPeriod periodTitle={ groupData[ LAST_30_DAYS_KEY ].label }
+				onHistoryItemDelete={ onHistoryItemDelete }
+				historyItems={ groupData[ LAST_30_DAYS_KEY ].items } /> );
+		}
+
+		for ( let i = 11; i >= 0; i-- ) {
+			if ( groupData[ i ] ) {
+				periods.push( <PromptHistoryPeriod periodTitle={ groupData[ i ].label }
+					onHistoryItemDelete={ onHistoryItemDelete }
+					historyItems={ groupData[ i ].items } /> );
+			}
+		}
+
+		return periods;
+	};
+
+	return (
+		<Modal
+			container={ () => document.querySelector( '.MuiDialogContent-root:not(.MuiBackdrop-root), #e-form-media .MuiBox-root' ) }
+			open={ true }
+			hideBackdrop={ true }
+			onClose={ onModalClose }
+			sx={ { position: 'absolute' } }
+			{ ...props }>
+			<Backdrop open={ true }
+				sx={ { position: 'absolute', justifyContent: 'flex-end', alignItems: 'flex-start' } }
+				aria-hidden={ false }>
+				<Slide direction="left" in={ true } mountOnEnter unmountOnExit>
+					<StyledContent aria-label={ __( 'Prompt history modal', 'elementor' ) }
+						fullHeight={ promptType === HISTORY_TYPES.IMAGE }>
+						<PromptHistoryModalHeader onClose={ onModalClose } />
+
+						{ error && NO_HISTORY_ACCESS_ERROR !== error && <PromptErrorMessage error={ error }
+							onRetry={ lastRun.current }
+							sx={ { position: 'absolute', zIndex: 1 } } /> }
+
+						{ isLoading && <LinearProgress
+							role="progressbar"
+							aria-label={ __( 'Loading', 'elementor' ) }
+							color="secondary" /> }
+
+						<Box sx={ { overflowY: 'scroll', height: '85%' } } ref={ scrollContainer }>
+							{ error && NO_HISTORY_ACCESS_ERROR === error && <PromptHistoryUpgrade variant="full" /> }
+
+							{ 0 === items?.length && <PromptHistoryEmpty promptType={ promptType } /> }
+
+							{ items?.length > 0 && <InfiniteScroll loadMore={ loadNext }
+								getScrollParent={ () => scrollContainer.current }
+								useWindow={ false }
+								isReverse={ false }
+								threshold={ 30 }
+								initialLoad={ false }
+								hasMore={ ! isLastPage }
+								children={ renderPeriods() } /> }
```

Comment:
> Why pass a children prop? 

Just a personal preference to keep it shorter as this:

```js
<InfiniteScroll loadMore={ loadNext }
		getScrollParent={ () => scrollContainer.current }
		useWindow={ false }
		isReverse={ false }
		threshold={ 30 }
		initialLoad={ false }
		hasMore={ ! isLastPage }
		children={ renderPeriods() } />;
```

instead of this:

```js
	<InfiniteScroll loadMore={ loadNext }
		getScrollParent={ () => scrollContainer.current }
		useWindow={ false }
		isReverse={ false }
		threshold={ 30 }
		initialLoad={ false }
		hasMore={ ! isLastPage } >
		{renderPeriods()}
	</InfiniteScroll>
```

> why pass the periods as a function and not as a component?

Semantically it doesn't sound like a component bc we just put items into the right periods here. So it's more like a more complex .map() call if you wish. Why do you think it should be a component in that case? 🤔